src/Npgsql/NpgsqlConnector.cs

```diff
@@ -488,7 +488,7 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
             }
         }
 
-        internal async Task LoadDatabaseInfo(bool forceReload, NpgsqlTimeout timeout, bool async,
+        internal async ValueTask LoadDatabaseInfo(bool forceReload, NpgsqlTimeout timeout, bool async,
```

Comment:
What is the point of this change? If the method returns asynchronously, the ValueTask will wrap a new Task so the heap allocation happens anyway. If the method returns synchronously (already cached), then a completed Task is returned (and not allocated).

Basically, non-generic ValueTask should never be used - except for very high-perf scenarios where IValueTaskSource is used.